### ISSUE TITLE
[link] Link does not fire "click" event from keyboard if it doesn't have an "href" attribute

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-design-system#10728]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Actual Behavior

`calcite-link` can be tabbed to, but doesn't respond to "enter" from keyboard and never fires "click", when there is no `href` property. This makes the component inaccessible when using a keyboard to navigate a page. 

### Expected Behavior

`calite-link` components should fire a "click" event when the user presses "enter" on keyboard regardless of if the component has a `href` attribute or not. There are legitimate use cases for a link component that just has onClick handlers, even if a `calcite-button` might be a more correct component.

### Reproduction Sample

https://codepen.io/geospatialem/pen/poMGXxd

### Reproduction Steps

1. Using the sample above (https://codepen.io/geospatialem/pen/poMGXxd) - thanks @geospatialem 
2. Open the codepen console
3. Click in the frame and tab to the first "Esri" link
4. Hit enter/return on keyboard
5. Notice no event fires

### Reproduction Version

2.13.2

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online
